src/module.c

```diff
@@ -6504,8 +6545,10 @@ void moduleHandleBlockedClients(void) {
     pthread_mutex_lock(&moduleUnblockedClientsMutex);
     /* Here we unblock all the pending clients blocked in modules operations
      * so we can read every pending "awake byte" in the pipe. */
-    char buf[1];
-    while (read(server.module_blocked_pipe[0],buf,1) == 1);
+    if (listLength(moduleUnblockedClients) > 0) {
+        char buf[1];
+        while (read(server.module_blocked_pipe[0],buf,1) == 1);
```

Comment:
Maybe I am missing something but wouldn't this will prevent you from reading from the pipe which will eventually cause the pipe to be full?

Reply:
If there is data in pipe, `listLength(moduleUnblockedClients)` should be greater than zero. I added if check here to just skip unnecessary `read()` call. If list is empty, pipe should be empty.

Reply:
Ohh sorry, you are reading if there is something in the list, so I believe you are good... :+1:

Reply:
I'm more concerned with `write()` calls actually, I can revert the change here for `read()` if this is confusing. Just added this check to skip one syscall in the critical path.